### PR TITLE
new module - bamstats generalstats

### PR DESCRIPTION
**Pull request by johnoooh**
## PR checklist

Closes #[3193](https://github.com/nf-core/modules/issues/3193) 

- [X] Adds [Bamstats](https://github.com/guigolab/bamstats/tree/master) from the Guigo lab
- [X] Added tests and they passed 

There are several tools named Bamstats, I have used this one before so I thought it fit to add.

### CODE REVIEW

File: modules/nf-core/bamstats/generalstats/main.nf [hunk 1]
```diff
@@ -0,0 +1,52 @@
+process BAMSTATS_GENERALSTATS {
+    tag "$meta.id"
+    label 'process_single'
+    conda "bioconda::bamstats=0.3.5"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bamstats:0.3.5--he881be0_0':
+        'biocontainers/bamstats:0.3.5--he881be0_0' }"
+
+    input:
+
```
Comment (ggabernet) [inline modules/nf-core/bamstats/generalstats/main.nf:10]:
```suggestion
```

File: tests/modules/nf-core/bamstats/generalstats/test.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## TODO nf-core: Please run the following command to build this file:
+#                nf-core modules create-test-yml /generalstats
```
Comment (ggabernet) [inline tests/modules/nf-core/bamstats/generalstats/test.yml:2]:
```suggestion
```

**Review by ggabernet — APPROVED** [review]
A couple of minor comments, but otherwise looks good to me!

**Comment by emi80** [comment]
Hi @johnoooh @ggabernet,

I created #3193 at the Hackaton in March last year. Now I see this PR (thanks @johnoooh). Do you guys think we can get this merged during this hackaton?

Many thanks!

Emilio

**Pull request merged**